java/code/src/com/redhat/rhn/frontend/action/common/DownloadFile.java

```diff
@@ -619,7 +619,18 @@ private StreamInfo getStreamInfoKickstart(ActionMapping mapping, ActionForm form
             // my $dp = File::Spec->catfile($kickstart_mount, $tree->base_path, $path);
 
             if (child == null) {
-                diskPath = kickstartMount + "/" + tree.getBasePath() + path;
+                if (path.contains("repodata/") && tree.getKernelOptions().contains("useonlinerepo")) {
+                    String[] split = StringUtils.split(path, '/');
+                    if (split[0].equals("repodata")) {
+                        split[0] = tree.getChannel().getLabel();
+                    }
+                    diskPath = "/var/cache/" +
```

Comment:
Does this correspond to the mount point for repomd cache? This is configurable (`repomd_cache_mount_point` global option). Maybe you can also pull the value out of config instead of hardcoding.

Reply:
Seems so. I just copied the code from some lines below.